### PR TITLE
Update version for runc compatibility for Moby

### PR DESCRIPTION
**Pull request by jhult**
Update version for compatibility with Moby. Fix for #529.

image

### CODE REVIEW

**Comment by utam0k** [comment]
Sorry, we are probably the cause of the CI failures 😭  Please wait a moment.

File: crates/youki/Cargo.toml [hunk 1]
```diff
@@ -31,3 +32,6 @@ clap_generate = { version = "3.0.0-beta.5" }
 [dev-dependencies]
 serial_test = "0.5.1"
 
+[build-dependencies]
+anyhow = "1.0"
```
Comment (utam0k) [inline crates/youki/Cargo.toml:36]:
```suggestion
anyhow = "1.0.51"
```

File: crates/youki/build.rs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+use anyhow::Result;
+use vergen::{Config, ShaKind, vergen};
+
+fn main() -> Result<()> {
```
Comment (utam0k) [inline crates/youki/build.rs:4]: Why do we need to use `anyhow::Result` in `build.rs`?

File: crates/youki/Cargo.toml (hunk 1 above)
Comment (jhult) [reply crates/youki/Cargo.toml:36]: Bumped this in `[build-dependencies]` and also bumped `anyhow` in the `[dependencies]` section.

File: crates/youki/build.rs (hunk 2 above)
Comment (jhult) [reply crates/youki/build.rs:4]: The [`vergen` function](https://docs.rs/vergen/latest/vergen/fn.vergen.html) returns a [`Result`](https://docs.rs/anyhow/1.0.51/anyhow/type.Result.html).

I based the code on [this](https://docs.rs/vergen/latest/vergen/#buildrs) example.

If there is another (or better) way to accomplish this, feel free to suggest it :)

Comment (utam0k) [reply crates/youki/build.rs:4]: Oh sorry, my point was ignorant.

**Comment by jhult** [comment]
Rebased on latest `main` and fixed formatting.

**Review by utam0k — APPROVED** [review]
Thanks!

**Pull request merged**